nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/QueuePrioritizer.java

```diff
@@ -18,74 +18,41 @@
+            .comparing(FlowFileRecord::getContentClaim, Comparator.nullsFirst(Comparator.naturalOrder()))
+            .thenComparingLong(FlowFileRecord::getContentClaimOffset);
+    private static final Comparator<FlowFileRecord> idComparator = Comparator.comparingLong(FlowFileRecord::getId);
+
+    private final transient List<FlowFilePrioritizer> prioritizers;
 
     public QueuePrioritizer(final List<FlowFilePrioritizer> priorities) {
-        if (null != priorities) {
-            prioritizers.addAll(priorities);
-        }
+        prioritizers = priorities == null ? List.of() : List.copyOf(priorities);
     }
 
     @Override
     public int compare(final FlowFileRecord f1, final FlowFileRecord f2) {
-        int returnVal = 0;
-        final boolean f1Penalized = f1.isPenalized();
-        final boolean f2Penalized = f2.isPenalized();
-
-        if (f1Penalized && !f2Penalized) {
-            return 1;
-        } else if (!f1Penalized && f2Penalized) {
-            return -1;
-        }
-
-        if (f1Penalized && f2Penalized) {
-            if (f1.getPenaltyExpirationMillis() < f2.getPenaltyExpirationMillis()) {
-                return -1;
-            } else if (f1.getPenaltyExpirationMillis() > f2.getPenaltyExpirationMillis()) {
-                return 1;
-            }
-        }
+        final int penaltyComparisonResult = penaltyComparator.compare(f1, f2);
+        if (penaltyComparisonResult != 0) return penaltyComparisonResult;
 
-        if (!prioritizers.isEmpty()) {
-            for (final FlowFilePrioritizer prioritizer : prioritizers) {
-                returnVal = prioritizer.compare(f1, f2);
-                if (returnVal != 0) {
-                    return returnVal;
-                }
-            }
+        for (FlowFilePrioritizer comparator : prioritizers) {
+            final int prioritizerComparisonResult = comparator.compare(f1, f2);
+            if (prioritizerComparisonResult != 0) return prioritizerComparisonResult;
         }
 
-        final ContentClaim claim1 = f1.getContentClaim();
-        final ContentClaim claim2 = f2.getContentClaim();
-
-
-        // put the one without a claim first
-        if (claim1 == null && claim2 != null) {
-            return -1;
-        } else if (claim1 != null && claim2 == null) {
-            return 1;
-        } else if (claim1 != null && claim2 != null) {
-            final int claimComparison = claim1.compareTo(claim2);
-            if (claimComparison != 0) {
-                return claimComparison;
-            }
-
-            final int claimOffsetComparison = Long.compare(f1.getContentClaimOffset(), f2.getContentClaimOffset());
-            if (claimOffsetComparison != 0) {
-                return claimOffsetComparison;
-            }
-        }
+        final int claimComparisionResult = claimComparator.compare(f1, f2);
+        if (claimComparisionResult != 0) return claimComparisionResult;
```

Comment:
As a general rule, braces should be used for conditional blocks:
```suggestion
        if (claimComparisionResult != 0) {
            return claimComparisionResult;
        }
```